### PR TITLE
Add section on flow run cancellation

### PR DESCRIPTION
**Pull request by bunchesofdonald**
Adds basic documentation on flow-run cancellation.

### CODE REVIEW

File: docs/concepts/flows.md [hunk 1]
```diff
@@ -609,5 +609,33 @@ Completed(message=None, type=COMPLETED, result='foo', flow_run_id=7240e6f5-f0a8-
...
+## Flow run cancellation
 
+Flow runs can be cancelled from the CLI, UI, REST API, or Python client. 
 
+When cancellation is requested, the flow run is moved to a "Cancelling" state. The agent monitors the state of flow runs and detects that cancellation has been requested. The agent then sends a signal to the flow run infrastructure, requesting termination of the run. If the run does not terminate after a grace period (default of 30 seconds), the infrastructure will be killed, ensuring the flow run exits.
+
+!!! warning "An agent is required"
+Flow run cancellation requires the flow run to be submitted by an agent and for an agent to be running to enforce the cancellation. Flow runs without deployments cannot be cancelled yet.
+
+Support for cancellation has been added to all core library infrastructure types:
```
Comment (madkinsz) [inline docs/concepts/flows.md:621]:
```suggestion
Support for cancellation is included for all core library infrastructure types:
```

File: docs/concepts/flows.md [hunk 2]
```diff
@@ -609,5 +609,33 @@ Completed(message=None, type=COMPLETED, result='foo', flow_run_id=7240e6f5-f0a8-
...
 
 
+## Flow run cancellation
 
+Flow runs can be cancelled from the CLI, UI, REST API, or Python client. 
 
+When cancellation is requested, the flow run is moved to a "Cancelling" state. The agent monitors the state of flow runs and detects that cancellation has been requested. The agent then sends a signal to the flow run infrastructure, requesting termination of the run. If the run does not terminate after a grace period (default of 30 seconds), the infrastructure will be killed, ensuring the flow run exits.
+
+!!! warning "An agent is required"
+Flow run cancellation requires the flow run to be submitted by an agent and for an agent to be running to enforce the cancellation. Flow runs without deployments cannot be cancelled yet.
```
Comment (tpdorsey) [inline docs/concepts/flows.md:619]:
```suggestion
    Flow run cancellation requires the flow run to be submitted by an agent and for an agent to be running to enforce the cancellation. Flow runs without deployments cannot be cancelled yet.
```

File: docs/concepts/flows.md [hunk 3]
```diff
@@ -609,5 +609,33 @@ Completed(message=None, type=COMPLETED, result='foo', flow_run_id=7240e6f5-f0a8-
...
+
+Support for cancellation has been added to all core library infrastructure types:
+
+- Docker Containers
+- Kubernetes Jobs
+- Processes
+
+### Via the CLI
```
Comment (tpdorsey) [inline docs/concepts/flows.md:627]:
```suggestion
### Cancel via the CLI
```

File: docs/concepts/flows.md [hunk 4]
```diff
@@ -609,5 +609,33 @@ Completed(message=None, type=COMPLETED, result='foo', flow_run_id=7240e6f5-f0a8-
...
+
+From the command line in your execution environment, you can cancel a flow run by using the `prefect flow-run cancel` CLI command, passing the ID of the flow run. 
+
+<div class="terminal">
+```bash
+$ prefect flow-run cancel 'a55a4804-9e3c-4042-8b59-b3b6b7618736'
+```
+</div>
+
+### Via the UI
```
Comment (tpdorsey) [inline docs/concepts/flows.md:637]:
```suggestion
### Cancel via the UI
```

**Review by tpdorsey — APPROVED** [review]

File: docs/concepts/flows.md [hunk 5]
```diff
@@ -624,20 +624,27 @@ Support for cancellation is included for all core library infrastructure types:
...
-    - The scope for Docker Containers is the Docker API's base url.
-    - The scope for Kubernetes Jobs is the name of the active Kubernetes cluster.
-    - The scope for Processes is the hostname as returned by `socket.gethostname`.
-- If the `pid` itself is not found the flow run will be marked as cancelled.
-    - The `pid` for Docker Containers is the id of the running container.
-    - The `pid` for Kubernetes Jobs is the name of the job.
-    - The `pid` for Processes is the process id.
-- If the agent runs into an unexpected error during cancellation the flow run may or may not be cancelled depending on where the error occured. The agent will try again to cancel the flow run.
+Cancellation is robust to restarts of the agent. To enable this, we track attach metadata about the created flow run infrastructure to the flow run. Internally, this is referred to as the `infrastructure_pid` or infrastructure identifier. Generally, this is composed of two parts:
```
Comment (madkinsz) [inline docs/concepts/flows.md]: Fixed this weird track/attach dup in the next commit.

**Pull request merged**